### PR TITLE
work around an issue starting atom from unix launchers

### PR DESCRIPTION
**Pull request by devoncarew**
Possible fix for https://github.com/flutter/flutter/issues/3531 - the paths aren't set up for atom when its started from the system launchers.

@danrubel, I don't have an easy way to test on a linux box - would you mind giving this a spin? W/o this patch, atom should fail to find adb or devices when started from the system launcher. W/ it, it should find devices. And this should still work when atom is started from the command line :)

### CODE REVIEW

File: lib/atom_utils.dart [hunk 1]
```diff
@@ -40,7 +40,9 @@ Future<String> which(String execName, {bool isBatchScript: false}) {
       return result;
     });
   } else {
-    return exec('which', [execName]).then((String result) {
+    if (_shellWrangler == null) _shellWrangler = new ShellWrangler();
```
Comment (danrubel) [inline lib/atom_utils.dart:36]: This looks identical to the (isMac) block above... combine with block above? if (!isWindows) ?

File: lib/node/process.dart [hunk 2]
```diff
@@ -16,6 +16,7 @@ final Process process = new Process._();
 final bool isWindows = process.platform.startsWith('win');
 final bool isMac = process.platform == 'darwin';
 final bool isLinux = !isWindows && !isMac;
+bool get isPosix => !isWindows;
```
Comment (danrubel) [inline lib/node/process.dart]: final like ones above?

Comment (devoncarew) [inline lib/node/process.dart]: will do

File: lib/atom_utils.dart (hunk 1 above)
Comment (devoncarew) [inline lib/atom_utils.dart:36]: Yeah, I think it is now - I'll consolidate.

File: lib/node/process.dart [hunk 3]
```diff
@@ -85,12 +86,14 @@ class ProcessRunner {
   factory ProcessRunner.underShell(String command, {
     List<String> args, String cwd, Map<String, String> env
   }) {
-    if (isMac && _shellWrangler == null) {
-      _shellWrangler = new MacShellWrangler();
-    }
+    if (isPosix && _shellWrangler == null) {
```
Comment (danrubel) [inline lib/node/process.dart]: Drop `&& _shellWrangler == null` from the if statement?

**Comment by danrubel** [comment]
LGTM. I can't test this at the moment, but will do so soon.

File: lib/node/process.dart (hunk 3 above)
Comment (devoncarew) [inline lib/node/process.dart]: oops!

**Comment by devoncarew** [comment]
Thanks! updated

**Comment by devoncarew** [comment]
fix https://github.com/dart-atom/atom.dart/issues/39

**Comment by danrubel** [comment]
I cannot reproduce the bug without the patch on my Linux box... Atom launches a test app on the device without any problem... thus I cannot confirm the fix.

**Pull request merged**